File: VTI.Windows/src/VTI.Core/QuesitoPreviewService.cs
namespace VTI.Core;

public sealed class QuesitoPreviewService
{
    private readonly LaTeXService _latex;
    private readonly PDFCompilerService _pdf;

    public QuesitoPreviewService(LaTeXService latex, PDFCompilerService pdf)
    {
        _latex = latex;
        _pdf = pdf;
    }

    public string CompilePdf(string latexBody)
    {
        var dir = Path.Combine(Path.GetTempPath(), "VTI-QuesitoPreview-" + Guid.NewGuid().ToString("N"));
        Directory.CreateDirectory(dir);
        var texPath = Path.Combine(dir, "QuesitoPreview.tex");
        _latex.WriteRenderedTemplate(LaTeXTemplate.QuesitoPreview, new Dictionary<string, string>
        {
            ["BODY"] = latexBody
        }, texPath);
        var result = _pdf.Compile(texPath);
        if (string.IsNullOrEmpty(result.PdfPath) || !File.Exists(result.PdfPath))
            throw AppException.PdfCompilationFailed(result.Log);
        return result.PdfPath;
    }
}

File: VTI/Services/QuesitoPreviewService.swift
import Foundation

struct QuesitoPreviewService: Sendable {
    let latexService: LaTeXService
    let pdfCompilerService: PDFCompilerService

    /// Compila un singolo blocco quesito (come nel `BODY` del test) con il template di anteprima.
    func compilePDF(latexBody: String) throws -> URL {
        let dir = FileManager.default.temporaryDirectory
            .appendingPathComponent("VTI-QuesitoPreview-\(UUID().uuidString)", isDirectory: true)
        try FileManager.default.createDirectory(at: dir, withIntermediateDirectories: true)

        let texURL = dir.appendingPathComponent("QuesitoPreview.tex")
        _ = try latexService.writeRenderedTemplate(
            template: .quesitoPreview,
            placeholders: [
                "BODY": latexBody
            ],
            destinationURL: texURL
        )

        let result = try pdfCompilerService.compile(texURL: texURL)
        guard let pdf = result.pdfURL else {
            throw AppError.pdfCompilationFailed(result.log)
        }
        return pdf
    }
}
